src/util/charbuf.c

```diff
@@ -31,9 +31,12 @@ void free_charbuf(charbuf * buf)
 {
   if (buf != NULL)
   {
-    free(buf->chars);
-    buf->chars = NULL;
-    buf->len = 0;
+    if(buf->len > 0)
```

Comment:
I know the charbuf class has checks for allocating a pointer of size 0, but someone could still do it themselves. While the behavior is undefined, the address or NULL returned should be freeable. What was the need to add this additional check?

Reply:
I don't think that extra check was necessary. I've removed it in 7de34c9.